src/regdesk/regdesk.js

```diff
@@ -25,10 +26,40 @@ function configureManager({
 }
 
 var printerMgr;
+var fontLoader = BadgeLabelBuilder.loadCustomFonts(document);
 
 document.addEventListener('readystatechange', async () => {
   if (document.readyState === "complete") {
     printerMgr = configureManager();
     await printerMgr.refreshPrinters();
+    await fontLoader;
```

Comment:
From line 29, yes.

To use the emoji font we need it to be loaded and available prior to attempting to use it in the canvas. I experimented with a few ways of loading fonts and ensuring they were available to both the page canvas elements as well as OffscreenCanvas elements. Explicitly loading it in JS and adding it to the Document's font list was the only _reliable_ way I could find to ensure this all happened. Other methods (CSS, hidden elements, etc) all had tradeoffs (hidden elements didn't actually load the font, CSS didn't load it into background canvas). 

Thus, we await loading the font. I'm not really happy with this solution, but it's expedient.

Reply:
I guess I'm just lost on it not really being shown as a method call. Can we add the `();` to make that clear?

Reply:
This might be a difference in how JS does async vs how I learned it. Is a promise a function you call? I thought the immediate return type of an async method is a promise _object_ that you then await in some fashion, not execute.

![image](https://user-images.githubusercontent.com/1441553/147442394-a92ee972-f021-43b9-8202-224dbecea749.png)

Reply:
Oooh, you're right. Didn't realize fontLoader was a promise.